src/Filesystem/JsonFileReader.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Kollex\Filesystem;
+
+use Assert\Assertion;
+use Kollex\Assortment\BaseProductPackagingType;
+use Kollex\Assortment\BaseProductUnitType;
+use Kollex\Assortment\PackagingType;
+use Kollex\Assortment\Product;
+use Kollex\Assortment\ProductInterface;
+
+class JsonFileReader implements FileReaderInterface
+{
+    /**
+     * @return ProductInterface[]
+     */
+    public function read(string $filename): iterable
+    {
+        $data = file_get_contents($filename);
```

Comment:
Need to process condition when we cannot read from a file like in your previous implementation.
`file_get_contents` can returns **false**